gzip.go

```diff
@@ -9,8 +10,49 @@ import (
 	"strings"
 )
 
+// ErrEmptyContentCoding indicates that the Accept-Encoding header was
+// empty.
+var ErrEmptyContentCoding = errors.New("Empty Accept-Encoding")
+
 type codings map[string]float64
 
+// KeyError is a Key/Value struct that matches the error with the
+// string that caused the error.
+type KeyError struct {
+	Key string // The "key", which is the ill-formatted string
+	Err error  // the "value", which is the error from strconv
+}
+
+func (k *KeyError) Error() string {
+	return fmt.Sprintf("'%s' caused error: '%v'", k.Key, k.Err.Error())
+}
+
+// ErrorList is a slice of KeyErrors that allows individual errors
+// to be extracted.
+type ErrorList []KeyError // Slice of returned errors
+
+func (e *ErrorList) Error() string {
+	return fmt.Sprintf("%d errors", len(*e))
+}
+
+// Append appends an error to the error list.
+func (e *ErrorList) Append(err interface{}) {
+	// Should this be a method on a pointer to a slice, or just
+	// a slice?
+	switch err := err.(type) {
+	case KeyError:
+		*e = append(*e, err)
+	case *KeyError:
+		*e = append(*e, *err)
+	case *ErrorList:
+		*e = append(*e, *err...)
+	case ErrorList:
+		*e = append(*e, err...)
```

Comment:
I don't think we need to handle these cases (e.g. appending another ErrorList) which don't appear in practice. This doesn't need to be a general-purpose interface, just enough functionality to convey our list of errors to the caller. Actually, since ErrorList can only contain KeyErrors anyway, it would be reasonable for this method to only accept those.